types/bluebird/bluebird-tests.ts

```diff
@@ -879,11 +882,18 @@ obj = Promise.promisifyAll(obj, {
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-const generator = function*(a: number, b: string) { return "string"; };
-const coroutine = Promise.coroutine<string, number, string>(generator);
-coroutine(5, "foo").then((x: string) => {});
+const generator1 = function*(a: number, b: string) { return "string"; };
+const coroutine1 = Promise.coroutine(generator1);
+strProm = coroutine1(5, "foo");
+
+const generator2 = function*(a: number, b: string) {
+	yield foo;
+	return bar;
+};
+const coroutine2 = Promise.coroutine(generator2);
```

Comment:
In the same vein, may I ask you to extend this test here cover the case when someone explicitly sets the type `T` of `.coroutine()`?
Because this is something that _must_ work at all times, even if you use `Promise.coroutine<Bar, number, string>`, and it'd be nice if this was covered in future PRs.

Reply:
It's fine if you _don't_ add this change to your PR. 🙂